crates/tests/src/rpc/read.rs

```diff
@@ -0,0 +1,1038 @@
+
+        curl --location 'https://free-rpc.nethermind.io/sepolia-juno/' \
+        --header 'Content-Type: application/json' \
+        --data '{
+            "jsonrpc": "2.0",
+            "method": "starknet_getClass",
+            "params": {
+                "block_id": {
+                    "block_number": 12
+                },
+                "class_hash": "0x05c478ee27f2112411f86f207605b2e2c58cdb647bac0df27f660ef2252359c6"
+            },
+            "id": 1
+        }'
+
+    In the test case, contract_class.json is same as juno.json used in the js script above.
+    */
+    #[tokio::test]
+    async fn test_get_class_works() {
+        let madara = get_shared_state().await;
+        let contract_class = {
+            let mut madara_guard = madara.lock().await;
+            madara_guard
+                .json_rpc()
+                .get_class(
+                    BlockId::Number(12),
+                    Felt::from_hex_unchecked("0x05c478ee27f2112411f86f207605b2e2c58cdb647bac0df27f660ef2252359c6"),
+                )
+                .await
+                .unwrap()
+        };
+
+        let contract_program = match contract_class.clone() {
+            ContractClass::Legacy(compressed) => Some(compressed.program),
+            _ => None,
+        };
+
+        let contract_entry_points = match contract_class.clone() {
+            ContractClass::Legacy(compressed) => Some(compressed.entry_points_by_type),
+            _ => None,
+        };
+
+        let contract_abi = match contract_class.clone() {
+            ContractClass::Legacy(compressed) => Some(compressed.abi),
+            _ => None,
+        };
+
+        let decompressed_program = decompress_to_string(contract_program.unwrap());
+
+        let mut class_program_file = File::open("src/rpc/test_utils/class_program.txt").unwrap();
+
+        let mut original_program = String::new();
+        class_program_file.read_to_string(&mut original_program).expect("issue while reading the file");
+
+        let contract_class_file = File::open("src/rpc/test_utils/contract_class.json").unwrap();
+        let reader = BufReader::new(contract_class_file);
+
+        let expected_contract_class: ContractClass = serde_json::from_reader(reader).unwrap();
+
+        let expected_contract_entry_points = match expected_contract_class.clone() {
+            ContractClass::Legacy(compressed) => Some(compressed.entry_points_by_type),
+            _ => None,
+        };
+
+        let expected_contract_abi = match expected_contract_class.clone() {
+            ContractClass::Legacy(compressed) => Some(compressed.abi),
+            _ => None,
+        };
+
+        assert_eq!(decompressed_program, original_program);
+        assert_eq!(contract_entry_points, expected_contract_entry_points);
+        assert_eq!(contract_abi, expected_contract_abi);
+    }
+
+    #[tokio::test]
+    async fn test_get_class_at_works() {
+        let madara = get_shared_state().await;
```

Comment:
a lot of common code between this and above, should we make an `assert_contract_class` function?

Reply:
done